hints/hint_types.py

```diff
@@ -0,0 +1,125 @@
+from dataclasses import dataclass
+from re import S
+from typing import List
+
+from logic.logic import Logic
+
+
+@dataclass
+class GossipStoneHint:
+    location: str
+    item: str
+    needs_logic: bool
+
+    def to_gossip_stone_text(self) -> List[str]:
+        """each string in the list appear in a separate textbox and will be line broken"""
+        raise NotImplementedError("abstract")
+
+    def to_spoiler_log_text(self) -> str:
+        raise NotImplementedError("abstract")
+
+    def __hash__(self):
+        return hash(self.location + self.item)
+
+
+@dataclass
+class GossipStoneHintWrapper:
+    primary_hint: GossipStoneHint
+    secondary_hint: GossipStoneHint
+
+    def to_gossip_stone_text(self) -> List[str]:
+        primary_text = self.primary_hint.to_gossip_stone_text()
+        secondary_text = self.secondary_hint.to_gossip_stone_text()
+        return [*primary_text, *secondary_text]
+
+    def to_spoiler_log_text(self) -> str:
+        return f"{self.primary_hint.to_spoiler_log_text()} / {self.secondary_hint.to_spoiler_log_text()}"
+
+
+@dataclass
+class TrialGateGossipStoneHint(GossipStoneHint):
+    trial_gate: str
+
+    def to_gossip_stone_text(self) -> List[str]:
+        return [
+            f"They say that opening the <r<{self.trial_gate}>> will reveal <y<{self.item}>>"
+        ]
+
+    def to_spoiler_log_text(self) -> str:
+        return f"{self.trial_gate} has {self.item}"
+
+    def __hash__(self):
+        return hash(self.location + self.item)
+
+
+@dataclass
+class LocationGossipStoneHint(GossipStoneHint):
+    location_string: str
+
+    def to_gossip_stone_text(self) -> List[str]:
+        return [f"They say that {self.location_string} <y<{self.item}>>"]
```

Comment:
```suggestion
        return [f"They say that <r<{self.location_string}>> <y<{self.item}>>"]
```
shouldn't the location part be colored like it used to be?

Reply:
This colors everything in the location string, including articles and the final verb (like reveals) which CJ thinks should not be highlighted (and I agree).

Reply:
hm that's unfortunate, the coloring makes it a lot easier to read imo but I agree as well that only the location name should be highlighted. Maybe we can revisit it in the future

Reply:
I'm currently adding coloring to every hint name